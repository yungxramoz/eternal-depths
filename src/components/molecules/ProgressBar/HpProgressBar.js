import React from 'react'
import RpgProgressBar from '../../atoms/RpgProgressBar/RpgProgressBar'
import './ProgressBar.css'

const HpProgressBar = ({ currentHp, maxHp }) => {
  if (currentHp < 0) {
    currentHp = 0
  }

  return (
    <div className="progress-container">
      <p className="hint-text">
        HP {currentHp}/{maxHp}
      </p>
      <RpgProgressBar max={maxHp} current={currentHp} type="hp" />
    </div>
  )
}

export default HpProgressBar
