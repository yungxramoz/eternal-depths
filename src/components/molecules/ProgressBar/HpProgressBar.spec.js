import { render, screen } from '@testing-library/react'
import React from 'react'
import HpProgressBar from './HpProgressBar'

describe('HpProgressBar', () => {
  it('renders correctly', () => {
    render(<HpProgressBar maxHp={100} currentHp={100} />)
    expect(document.body).toMatchSnapshot('HpProgressBar')
  })

  it('displays correct HP values', () => {
    render(<HpProgressBar maxHp={100} currentHp={50} />)
    expect(screen.getByText('HP 50/100')).toBeInTheDocument()
  })
})
