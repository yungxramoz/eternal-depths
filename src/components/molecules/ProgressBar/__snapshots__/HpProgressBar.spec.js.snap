// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HpProgressBar renders correctly: HpProgressBar 1`] = `
<body>
  <div>
    <div
      class="progress-container"
    >
      <p
        class="hint-text"
      >
         HP 
        100
        /
        100
      </p>
      <div
        class="rpgui-progress"
        data-rpguitype="progress"
      >
        <div
          class=" rpgui-progress-track"
        >
          <div
            class="rpgui-progress-fill red progress-animation"
            style="left: 0px; width: 100%;"
          />
        </div>
        <div
          class=" rpgui-progress-left-edge"
        />
        <div
          class=" rpgui-progress-right-edge"
        />
      </div>
    </div>
  </div>
</body>
`;
